### ISSUE TITLE
Cannot compile Beam binaries for Raspberry PI 3B+

### ISSUE THREAD

**Issue by tgbv** [BeamMW/beam#1823]
I cannot compile Beam binaries for a Raspberry PI 3B+ model. 

**To Reproduce**
1. Installed all required dependencies to their latest versions.
2. Do `export CC=gcc-8 && export CXX=g++-8`
3. Do `cmake -DCMAKE_BUILD_TYPE=Release && make -j 4`

**Platform and build**
* Linux raspberrypi 5.10.17-v7 SMP Fri Apr 30 13:18:35 BST 2021 armv7l GNU/Linux
* Beam version 6.1.12023

**Bash log**
```
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


On branch mainnet
Your branch is up to date with 'origin/mainnet'.

nothing to commit, working tree clean
BRANCH_NAME: mainnet
BEAM_VERSION: 6.1.12023
start libuv
Linux
GNU
-fvisibility=hidden;--std=gnu89;-Wall;-Wextra;-Wstrict-prototypes;-Wno-unused-parameter
-- Configuring done
-- Generating done
-- Build files have been written to: /media/pi/Intenso/beam
Consolidate compiler generated dependencies of target secp256k1
Consolidate compiler generated dependencies of target pbkdf
Consolidate compiler generated dependencies of target keccak
[  1%] Built target keccak
[  1%] Building CXX object 3rdparty/libbitcoin/src/wallet/CMakeFiles/libbitcoin_wallet.dir/bitcoin_uri.cpp.o
g++-8: error: unrecognized -march target: core2
g++-8: note: valid arguments are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5t armv5e armv5te armv5tej armv6 armv6j armv6k armv6z armv6kz armv6zk armv6t2 armv6-m armv6s-m armv7 armv7-a armv7ve armv7-r armv7-m armv7e-m armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8-m.base armv8-m.main armv8-r iwmmxt iwmmxt2 native
g++-8: error: missing argument to ‘-march=’
make[2]: *** [3rdparty/libbitcoin/src/wallet/CMakeFiles/libbitcoin_wallet.dir/build.make:76: 3rdparty/libbitcoin/src/wallet/CMakeFiles/libbitcoin_wallet.dir/bitcoin_uri.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1622: 3rdparty/libbitcoin/src/wallet/CMakeFiles/libbitcoin_wallet.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  2%] Built target pbkdf
[  2%] Built target secp256k1
[  2%] Building CXX object 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/chain_state.cpp.o
[  2%] Building CXX object 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/compact.cpp.o
[  2%] Building CXX object 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/block.cpp.o
[  3%] Building CXX object 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/header.cpp.o
g++-8: error: unrecognized -march target: core2
g++-8: note: valid arguments are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5t armv5e armv5te armv5tej armv6 armv6j armv6k armv6z armv6kz armv6zk armv6t2 armv6-m armv6s-m armv7 armv7-a armv7ve armv7-r armv7-m armv7e-m armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8-m.base armv8-m.main armv8-r iwmmxt iwmmxt2 native
g++-8: error: missing argument to ‘-march=’
make[2]: *** [3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/build.make:90: 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/chain_state.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
g++-8: error: unrecognized -march target: core2
g++-8: error: unrecognized -march target: core2
g++-8: note: valid arguments are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5t armv5e armv5te armv5tej armv6 armv6j armv6k armv6z armv6kz armv6zk armv6t2 armv6-m armv6s-m armv7 armv7-a armv7ve armv7-r armv7-m armv7e-m armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8-m.base armv8-m.main armv8-r iwmmxt iwmmxt2 native
g++-8: note: valid arguments are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5t armv5e armv5te armv5tej armv6 armv6j armv6k armv6z armv6kz armv6zk armv6t2 armv6-m armv6s-m armv7 armv7-a armv7ve armv7-r armv7-m armv7e-m armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8-m.base armv8-m.main armv8-r iwmmxt iwmmxt2 native
g++-8: error: missing argument to ‘-march=’
g++-8: error: missing argument to ‘-march=’
make[2]: *** [3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/build.make:76: 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/block.cpp.o] Error 1
make[2]: *** [3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/build.make:104: 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/compact.cpp.o] Error 1
g++-8: error: unrecognized -march target: core2
g++-8: note: valid arguments are: armv2 armv2a armv3 armv3m armv4 armv4t armv5 armv5t armv5e armv5te armv5tej armv6 armv6j armv6k armv6z armv6kz armv6zk armv6t2 armv6-m armv6s-m armv7 armv7-a armv7ve armv7-r armv7-m armv7e-m armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8-m.base armv8-m.main armv8-r iwmmxt iwmmxt2 native
g++-8: error: missing argument to ‘-march=’
make[2]: *** [3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/build.make:118: 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/header.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1388: 3rdparty/libbitcoin/src/chain/CMakeFiles/libbitcoin_chain.dir/all] Error 2
make: *** [Makefile:101: all] Error 2
```

I'm not a C++ guru. Maybe support for ARM processors must be inserted in configuration files?

**Comment by tgbv** [comment]
Modified `CMAKE_CXX_FLAGS`, have set march to `armv7` and it seems to compile now, but it's very slow.

**Comment by tgbv** [comment]
Compilation with 4 concurrent processes crashes the PI. Compilation with one process seems to work but is still very slow (`make -j 1`)

I received a warning treated as error during compilation processs:
```
[ 51%] Building CXX object keykeeper/CMakeFiles/keykeeper.dir/remote_key_keeper.cpp.o
/media/pi/Intenso/beam/keykeeper/remote_key_keeper.cpp: In member function ‘virtual void beam::wallet::RemoteKeyKeeper::InvokeAsync(beam::wallet::IPrivateKeyKeeper2::Method::CreateVoucherShielded&, const Ptr&)’:
/media/pi/Intenso/beam/keykeeper/remote_key_keeper.cpp:201:13: error: ‘dummy’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
       memcpy(&buf.front(), &x, sizeof(T));
       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [keykeeper/CMakeFiles/keykeeper.dir/build.make:90: keykeeper/CMakeFiles/keykeeper.dir/remote_key_keeper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3082: keykeeper/CMakeFiles/keykeeper.dir/all] Error 2
make: *** [Makefile:101: all] Error 2
```

**Comment by tgbv** [comment]
I've compiled the mainnet successfully.  For raspberry PI a separate tutorial section should be created.

1. For successful compilation we need at least 2GB of RAM. If PI device has only 1GB, an additional 1GB of swap space is required, preferable on a device with fast seek time. I used an external SSD. Avoid using the SD card on which OS is installed.

2. Cmake version from raspbian repositories is usually not the latest one. I had to download the source code (https://cmake.org/download/) and build it for armv7 myself.

3. `CMakeLists.txt` must configure the target architecture to `armv7` and not `core2`

4. The warning from https://github.com/BeamMW/beam/issues/1823#issuecomment-917552862 should either be fixed or suppressed by configuring `CMakeLists.txt` to not treat warnings as errors.

5. You must use only one make job. More can crash the PI device (`make -j 1`)
